dao/src/main/java/com/epam/esm/dao/impl/GiftCertificateDaoImpl.java

```diff
@@ -0,0 +1,96 @@
+package com.epam.esm.dao.impl;
+
+import com.epam.esm.dao.GiftCertificateDao;
+import com.epam.esm.dao.constant.sql.GiftCertificateSql;
+import com.epam.esm.dao.mapper.GiftCertificateMapper;
+import com.epam.esm.entity.GiftCertificate;
+import com.epam.esm.error.ExceptionCauseCode;
+import com.epam.esm.exception.ResourceNotAddedException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.support.GeneratedKeyHolder;
+import org.springframework.jdbc.support.KeyHolder;
+import org.springframework.stereotype.Repository;
+
+import java.math.BigDecimal;
+import java.sql.PreparedStatement;
+import java.sql.Statement;
+import java.sql.Timestamp;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * GiftCertificateDao implementation.
+ *
+ * @author Yauheni Tsitov
+ */
+@Repository
+public class GiftCertificateDaoImpl implements GiftCertificateDao {
+    private final JdbcTemplate jdbcTemplate;
+    private final GiftCertificateMapper mapper;
+
+    @Autowired
+    public GiftCertificateDaoImpl(JdbcTemplate jdbcTemplate, GiftCertificateMapper mapper) {
+        this.jdbcTemplate = jdbcTemplate;
+        this.mapper = mapper;
+    }
+
+    @Override
+    public long add(GiftCertificate giftCertificate) {
```

Comment:
I think you should return a created object, not `long`

Reply:
Why would I return the same object that I added, only additionally with the id field filled?

Reply:
Mistake.Return GiftCertificate